util.h

```diff
@@ -21,6 +21,8 @@
 
 #define ERRSTR (strerror (errno))
 
+#define DMON_EOF (-2)
```

Comment:
Using this constant looks clunky to me. End-of-file is not technically an error, that's why zero is returned in that case, and the code here is using the POSIX convention of returning a negative number to indicate an error—it could be any negative number. But the function did in fact read one byte (the delimiter) so returning zero (which means “end of file”) is not quite right.

A better approach would be to make `freaduntil()` include the delimiter in the resulting buffer, and if the returned buffer has `dbuf_size(&linebuf) == 1` then that means that only the delimiter has been put in the buffer. As a bonus, including the new line in the buffer means that it does not need to be added later to the output by hand.

Reply:
> A better approach would be to make...

Great suggestion. I like this solution a lot better! I have updated this PR with that changes. This PR will now fix #10 and #11.